test/src/lib/update-lock.spec.ts

```diff
@@ -0,0 +1,272 @@
+import { expect } from 'chai';
+import { SinonSpy, SinonStub, spy, stub } from 'sinon';
+import * as path from 'path';
+import * as Bluebird from 'bluebird';
+
+import rewire = require('rewire');
+import mockFs = require('mock-fs');
+
+import * as constants from '../../../src/lib/constants';
+import { UpdatesLockedError } from '../../../src/lib/errors';
+
+describe('lib/update-lock', () => {
+	const updateLock = rewire('../../../src/lib/update-lock');
+	const breadcrumbFiles = [
+		'rollback-health-breadcrumb',
+		'rollback-altboot-breadcrumb',
+	];
+
+	const mockBreadcrumbs = (...breadcrumbs: string[]) => {
+		const breadcrumbDir: { [key: string]: any } = {};
+		breadcrumbs.forEach((file) => (breadcrumbDir[file] = ''));
+		mockFs({
+			[path.join(
+				constants.rootMountPoint,
+				constants.stateMountPoint,
+			)]: breadcrumbDir,
+		});
+	};
+
+	const mockLockfile = (appId: number, service: string) => {
+		mockFs({
+			[path.join(
+				constants.rootMountPoint,
+				updateLock.lockPath(appId),
+				service,
+			)]: {
+				'updates.lock': '',
+			},
+		});
+	};
+
+	// TODO: Remove these hooks when we don't need './test/data' as test process's rootMountPoint
```

Comment:
These `before`/`after` hooks can be removed when every test that interacts with the fs in some way is switched over to mock-fs instead of using the physical file system.